git.go

```diff
@@ -9,12 +9,23 @@ import (
 	"io"
 	"io/ioutil"
 	"strconv"
+	"strings"
 
+	blocks "github.com/ipfs/go-block-format"
 	cid "github.com/ipfs/go-cid"
-	node "github.com/ipfs/go-ipld-node"
+	node "github.com/ipfs/go-ipld-format"
 	mh "github.com/multiformats/go-multihash"
 )
 
+func init() {
+	// Register the block decoder
+	node.DefaultBlockDecoder[cid.GitRaw] = DecodeBlock
+}
+
+func DecodeBlock(block blocks.Block) (node.Node, error) {
+	return ParseObjectFromBuffer(block.RawData())
```

Comment:
I'm thinking this should return an error if `block.Cid().Prefix() != { SHA1, Cid#1, GitRaw }`. Basically, we have the following options:

1. Pass through the CID to ensure it stays the same. This means `someGitNode.Cid()` could return a non-SHA1 CID.
2. Allow `block.Cid() != DecodeBlock(block).Cid()`. I really want to keep this invariant. IMO, a `Node` should be a "view" into a `Block` but shouldn't actually change its identity (CID).
3. Refuse to parse git blocks with non SHA1 CIDs (my proposal).

Thoughts?